lib/Open/index.js

```diff
@@ -1,100 +1,134 @@
-      stream : function(offset, length) {
+      stream: function (offset, length) {
         const options = Object.create(params);
-        const end = length ? offset + length : '';
+        const end = length ? offset + length : "";
         options.headers = Object.create(params.headers);
-        options.headers.range = 'bytes='+offset+'-' + end;
+        options.headers.range = "bytes=" + offset + "-" + end;
         return request(options);
       },
-      size: function() {
-        return new Promise(function(resolve, reject) {
+      size: function () {
+        return new Promise(function (resolve, reject) {
           const req = request(params);
-          req.on('response', function(d) {
-            req.abort();
-            if (!d.headers['content-length'])
-              reject(new Error('Missing content length header'));
-            else
-              resolve(d.headers['content-length']);
-          }).on('error', reject);
+          req
+            .on("response", function (d) {
+              req.abort();
+              if (!d.headers["content-length"])
+                reject(new Error("Missing content length header"));
+              else resolve(d.headers["content-length"]);
+            })
+            .on("error", reject);
         });
-      }
+      },
     };
 
     return directory(source, options);
   },
 
-  s3 : function(client, params, options) {
+  s3: function (client, params, options) {
     const source = {
-      size: function() {
-        return new Promise(function(resolve, reject) {
-          client.headObject(params, function(err, d) {
-            if (err)
-              reject(err);
-            else
-              resolve(d.ContentLength);
+      size: function () {
+        return new Promise(function (resolve, reject) {
+          client.headObject(params, function (err, d) {
+            if (err) reject(err);
+            else resolve(d.ContentLength);
           });
         });
       },
-      stream: function(offset, length) {
+      stream: function (offset, length) {
         const d = {};
-        for (const key in params)
-          d[key] = params[key];
-        const end = length ? offset + length : '';
-        d.Range = 'bytes='+offset+'-' + end;
+        for (const key in params) d[key] = params[key];
+        const end = length ? offset + length : "";
+        d.Range = "bytes=" + offset + "-" + end;
         return client.getObject(d).createReadStream();
-      }
+      },
     };
 
     return directory(source, options);
   },
+  s3_v3: function (client, params, options) {
+    const {
+      GetObjectCommand,
+      HeadObjectCommand,
+    } = require("@aws-sdk/client-s3");
 
-  custom: function(source, options) {
+    const source = {
+      size: async () => {
+        const head = await client.send(
+          new HeadObjectCommand({
+            Bucket: params.Bucket,
+            Key: params.Key,
+          })
+        );
+
+        return head.ContentLength ?? 0;
```

Comment:
The ?? operator is not supported in node v10.x and v11.x - you need to change to ||
(see tests breaking https://github.com/ZJONSSON/node-unzipper/actions/runs/9831777430/job/27160265748)

Please run unit tests for all supported versions, either by enabling github actions on your fork or by testing locally